### PR TITLE
Introduce function compilation

### PR DESCRIPTION
**Pull request by dvargas92495**
This PR introduces the utility for function compilation into our Vellum-native `FunctionDefinition` class. We simply solve the base case, and will be rolling out each test case per PR. We eventually (the last PR) will use this utility within our inline prompt nodes to allow users to pass in arbitrary functions defined locally as arguments.

### CODE REVIEW

File: src/vellum/workflows/utils/functions.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from typing import Callable
+
+from vellum.client.types.function_definition import FunctionDefinition
+
+
+def compile_function_definition(function: Callable) -> FunctionDefinition:
```
Comment (dvargasfuertes) [inline src/vellum/workflows/utils/functions.py:6]: I debated whether I should start the `src/vellum/prompts` module and put this in `src/vellum/prompts/utils/functions.py` instead. Thoughts?

File: src/vellum/workflows/utils/tests/test_functions.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from vellum.client.types.function_definition import FunctionDefinition
+from vellum.workflows.utils.functions import compile_function_definition
+
+
+def test_compile_function_definition__just_name():
```
Comment (dvargasfuertes) [inline src/vellum/workflows/utils/tests/test_functions.py:5]: More tests coming in future PRs

**Comment by marlenc3** [comment]
✅

File: src/vellum/workflows/utils/functions.py (hunk 1 above)
Comment (noanflaherty) [reply src/vellum/workflows/utils/functions.py:6]: I'm game!

**Review by noanflaherty — COMMENTED** [review]
I love where this is headed. I have some spicy ideas I wanna run by you

**Review by kirawoods — COMMENTED** [review]
✅

**Pull request merged**